testing/execution.ts

```diff
@@ -951,3 +941,151 @@ class DeveloperError extends Error {
+      );
+    }
+    const response: GenericSyncFormulaResult = vm
+      ? await executeFormulaOrSyncWithRawParamsInVM({
+          formulaSpecification,
+          params,
+          bundleSourceMapPath,
+          bundlePath,
+          executionContext,
+        })
+      : await executeFormulaOrSyncWithRawParams({formulaSpecification, params, manifest, executionContext});
+
+    result.push(...response.result);
+    if (contextOptions?.syncOptions?.includePermissions) {
+      // Denormalize the sync result before passing back into executeGetPermissions
+      const schema =
+        executionContext.sync.schema ??
+        findSyncFormula(manifest, formulaSpecification.formulaName, executionContext.authenticationName)?.schema;
+      const denormalizedSyncResult = untransformBody(response.result, schema);
+
+      rowAccessDefinitions.push(
+        ...(await getPermissionsForRows({
+          rows: denormalizedSyncResult,
+          formulaSpecification,
+          params,
+          manifest,
+          executionContext,
+          vm,
+          bundleSourceMapPath,
+          bundlePath,
+        })),
+      );
+    }
+    executionContext.sync.continuation = response.continuation;
+    iterations++;
+  } while (executionContext.sync.continuation && result.length < maxRows);
+  if (result.length > maxRows) {
+    result = result.slice(0, maxRows);
+  }
+  if (rowAccessDefinitions.length > maxRows) {
+    rowAccessDefinitions = rowAccessDefinitions.slice(0, maxRows);
+  }
+  return {result, rowAccessDefinitions};
+}
+
+/**
+ * This function handles running the GetPermissions formula for a given set of rows.
+ * Rows should be denormalized before being passed in.
+ *
+ * @param rows The rows to get permissions for
+ * @param formulaSpecification The Sync formula specification that is driving the rows we want to get permissions for.
+ * @param params The params to pass to the formula
+ * @param manifest The manifest of the pack
+ * @param executionContext The execution context
+ * @param vm Whether to run in a VM
+ * @param bundleSourceMapPath The source map path
+ * @param bundlePath The bundle path
+ * @returns Returns an array of the rowAccessDefinitions
+ */
+async function getPermissionsForRows({
+  rows,
+  formulaSpecification,
+  params,
+  manifest,
+  executionContext,
+  vm,
+  bundleSourceMapPath,
+  bundlePath,
+}: {
+  rows: any[];
+  formulaSpecification: SyncFormulaSpecification;
+  params: string[];
+  manifest: BasicPackDefinition;
+  executionContext: SyncExecutionContext;
+  vm?: boolean;
+  bundleSourceMapPath: string;
+  bundlePath: string;
+}) {
+  const getPermissionsFormulaSpecification: GetPermissionsFormulaSpecification = {
+    ...formulaSpecification,
+    type: FormulaType.GetPermissions,
+  };
+  const mappedRows = rows.map(row => ({row}));
+  const getPermissionsParams = [...params, JSON.stringify({rows: mappedRows})];
+
+  const response: any = vm
```

Comment:
Again no need to block this PR, but eventually this should do a loop where the rows are batched into arrays of size `maxPermissionBatchSize`.